src/AutoMapper/Execution/TypeMapPlanBuilder.cs

```diff
@@ -364,7 +364,8 @@ private Expression CreateConstructorParameterExpression(ConstructorParameterMap
         }
 
         private Expression TryPropertyMap(PropertyMap propertyMap)
-        {
+        {
+            propertyMap.CheckMappedReadonly();

```

Comment:
I thought about configuration time, but it would have to be in many places, it doesn't seem practical. So just move it in TypeMap.Seal before building the expression?

Reply:
I've done it after the rest of the configuration.